### PR TITLE
Update of mafft wrapper and addition of mafft --add tool

### PR DESCRIPTION
**Pull request by Delphine-L**
@davebx  is moving the 'FASTA' dependency from the biobuilds repo to the bioconda one, and then the tests should pass.

### CODE REVIEW

File: tools/mafft/mafft-add.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<tool id="rbc_mafft" name="MAFFT" version="@TOOL_VERSION@+galaxy@VERSION@">
```
Comment (bgruening) [inline tools/mafft/mafft-add.xml:2]: The id needs to be rbc_mafft_add?

File: tools/mafft/mafft-add.xml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      #if $sequences.sequenceType == 'singleseq'
+        $sequences.preservegap  $inputSequences
+      #elif $sequences.sequenceType == 'frags'
+        --addfragments $inputSequences
+      #elif $sequences.sequenceType == 'group'
+        --addprofile $inputSequences
+      #end if
+      $keeplength $map $reorder
+      $inputAlignment
+      > $outputAlignment
```
Comment (bgruening) [inline tools/mafft/mafft-add.xml:28]: all input and output files needs to be single quoted

File: tools/mafft/mafft-add.xml (hunk 1 above)
Comment (bgruening) [reply tools/mafft/mafft-add.xml:2]: Also the name and the description needs to be adopted

File: tools/mafft/mafft-add.xml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    mafft --version
+    ]]>
+  </version_command>
+  <command>
+    <![CDATA[
+      mafft
+      #if $sequences.sequenceType == 'singleseq'
+        $sequences.preservegap  $inputSequences
+      #elif $sequences.sequenceType == 'frags'
+        --addfragments $inputSequences
```
Comment (bgruening) [inline tools/mafft/mafft-add.xml:22]: single quotes

File: tools/mafft/mafft-add.xml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    <param name="reorder" type="boolean" truevalue="" falsevalue="--reorder" checked="True" label="Preserve the original order of sequences (--reorder)"  />
+  </inputs>
+  <outputs>
+    <data format="fasta" name="outputAlignment" label="${tool.name} on ${on_string} : New alignment"/>
+    <data name="outputmap" format="tabular" label="${tool.name} on ${on_string} : Correspondance of positions ">
+      <filter>map == True</filter>
+    </data>
+  </outputs>
+  <tests>
+    <test>
```
Comment (bgruening) [inline tools/mafft/mafft-add.xml:63]: can you add here the expected_num_outputs ... https://docs.galaxyproject.org/en/latest/dev/schema.html#id78

File: tools/mafft/mafft-add.xml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      #elif $sequences.sequenceType == 'frags'
+        --addfragments $inputSequences
+      #elif $sequences.sequenceType == 'group'
+        --addprofile $inputSequences
+      #end if
+      $keeplength $map $reorder
+      $inputAlignment
+      > $outputAlignment
+      #if $map == '--mapout'
+        && mv ${inputSequences}.map $outputmap
```
Comment (bgruening) [inline tools/mafft/mafft-add.xml:30]: single quoted

File: tools/mafft/mafft.xml [hunk 6]
```diff
@@ -1,229 +1,289 @@
...
+    <exit_code range="1:" level="fatal" description="Error occurred. Please check Tool Standard Error" />
+    <exit_code range=":-1" level="fatal" description="Error occurred. Please check Tool Standard Error" />
+  </stdio>
+  <version_command>    <![CDATA[
+    mafft --version
+    ]]>
+  </version_command>
+  <command>
+    <![CDATA[
+      export FASTA_4_MAFFT=`which fasta36`;
```
Comment (bgruening) [inline tools/mafft/mafft.xml:18]: Why is this needed?

File: tools/mafft/mafft-add.xml (hunk 2 above)
Comment (bgruening) [reply tools/mafft/mafft-add.xml:28]: ping

File: tools/mafft/mafft.xml (hunk 6 above)
Comment (Delphine-L) [reply tools/mafft/mafft.xml:18]: Mafft ask for the fasta3x path to be exported in FASTA_4_MAFFT to use the --fastapair option. I also could export it only if the --fastapair option is selected..

**Comment by bgruening** [comment]
Thanks a lot!
